packages/shell-api/src/shell-bson.spec.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint @typescript-eslint/camelcase: 0, new-cap: 0, @typescript-eslint/ban-ts-ignore: 0 */
+import shellBson from './shell-bson';
+import { expect } from 'chai';
+
+const hex_1234 = '31323334';
+const b64_1234 = 'MTIzNA==';
+const utf_1234 = '1234';
+
+describe('Shell BSON', () => {
+  describe('Code', () => {
+    it('expects arguments in order', () => {
+      const code = shellBson.Code('code', { k: 'v' });
+      expect(code.code).to.equal('code');
+      expect(code.scope).to.deep.equal({ k: 'v' });
+    });
+  });
+  describe('Date', () => {
+    it('returns string without new', () => {
+      expect(shellBson.Date()).to.be.a('string');
+    });
+    it('returns now object with new', () => {
+      // @ts-ignore
+      const date = new shellBson.Date();
+      const cDate = new Date();
+      expect(typeof date).to.equal('object');
+      // @ts-ignore
+      expect(date.getYear()).to.equal(cDate.getYear());
```

Comment:
`getYear` is deprecated, not sure if that is necessary for this test, but you can replace it with `getFullYear`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Deprecated_and_obsolete_features